sdcm/cluster.py

```diff
@@ -2572,24 +2573,30 @@ def restart_node_with_resharding(self, murmur3_partitioner_ignore_msb_bits: int
         search_reshard_finish = self.follow_system_log(patterns=[DB_LOG_PATTERN_RESHARDING_FINISH])
         start_scylla_timeout = 7200
         self.start_scylla(timeout=start_scylla_timeout)
-        resharding_started = list(search_reshard_start)
-        resharding_finished = list(search_reshard_finish)
 
-        if resharding_started:
-            self.log.debug(f'Resharding has been started successfully '
-                           f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits})')
-        else:
-            raise Exception(f'Resharding has not been started '
-                            f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits}) '
-                            'Check the log for the details')
+        def wait_resharding_log():
 
-        if resharding_finished:
-            self.log.debug(f'Resharding has been finished successfully '
-                           f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits})')
-        else:
-            raise Exception(f'Resharding has not been finished within {start_scylla_timeout}'
-                            f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits}) '
-                            'Check the log for the details')
+            resharding_started = list(search_reshard_start)
+            resharding_finished = list(search_reshard_finish)
+
+            if resharding_started:
+                self.log.debug(f'Resharding has been started successfully '
+                               f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits})')
+            else:
+                raise Exception(f'Resharding has not been started '
+                                f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits}) '
+                                'Check the log for the details')
+
+            if resharding_finished:
+                self.log.debug(f'Resharding has been finished successfully '
+                               f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits})')
+            else:
+                raise Exception(f'Resharding has not been finished within {start_scylla_timeout}'
+                                f'(murmur3_partitioner_ignore_msb_bits={murmur3_partitioner_ignore_msb_bits}) '
+                                'Check the log for the details')
+            return True
+
+        wait_for(wait_resharding_log, step=5, timeout=900, text="Wait resharding log messages...")
```

Comment:
maybe the `wait_for_log_lines` context manager is a better fit for this case ? 

what do you think @soyacz

Reply:
yes, I think this is exactly the case for `wait_for_log_lines`
@aleksbykov you can enhance this method with another info about context that could be appended to error log messages in it and also add debug messages in some places.